compiler/rustc_borrowck/src/diagnostics/region_name.rs

```diff
@@ -258,8 +258,8 @@ impl<'tcx> MirBorrowckCtxt<'_, 'tcx> {
             .or_else(|| self.give_name_if_anonymous_region_appears_in_output(fr))
             .or_else(|| self.give_name_if_anonymous_region_appears_in_yield_ty(fr))
             .or_else(|| self.give_name_if_anonymous_region_appears_in_impl_signature(fr))
-            .or_else(|| self.give_name_if_anonymous_region_appears_in_arg_position_impl_trait(fr));
-
+            .or_else(|| self.give_name_if_anonymous_region_appears_in_arg_position_impl_trait(fr))
+            .or_else(|| self.give_name_if_anonymous_region_appears_in_early_param(fr));
```

Comment:
Not sure if this is the right approach, since I don't ever think we expect this to happen unless we have a pre-existing error.

Can we just handle the unwrap in #118021 more responsibly?

Reply:
My initial solution was that but it felt kinda hacky, will do something similar thanks for suggestion!

Reply:
If you think it's hacky, then I can probably leave more feedback when I can see the changes implemented. I just don't think we should implement something that looks like "good-path" code that we only expect to implement when errors are co-occurring.